tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/AbstractTaggedMetricRegistry.java

```diff
@@ -74,7 +77,8 @@ protected Supplier<Counter> counterSupplier() {
     @Nonnull
     @SuppressWarnings("NoFunctionalReturnType") // metric factory
     protected Supplier<Histogram> histogramSupplier() {
-        return () -> new Histogram(createReservoir());
+        return () ->
+                new HistogramWithSamples(createReservoir(), new LockFreeSampler(SamplesTracer::getTraceIfObservable));
```

Comment:
Can you write up some benchmarks and napkin math on the impacts of capturing 1 (to N) samples per histogram? We often have *many* Histograms (including in timers), so want to understand memory footprint as well as concurrency